services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -186,15 +187,20 @@ async def start_project_interactive_services(
 
 
 async def delete_project(app: web.Application, project_uuid: str, user_id: int) -> None:
-    await _delete_project_from_db(app, project_uuid, user_id)
+    """
 
-    async def _remove_services_and_data():
-        await remove_project_interactive_services(
-            user_id, project_uuid, app, notify_users=False
-        )
-        await storage_api.delete_data_folders_of_project(app, project_uuid, user_id)
+    raises ProjectDeleteError
+    """
+    try:
+        await _delete.mark_project_as_deleted(app, project_uuid)
 
-    fire_and_forget_task(_remove_services_and_data())
+    except ProjectNotFoundError as err:
+        raise ProjectDeleteError(project_uuid, reason=f"Invalid project {err}") from err
+
+    if not _delete.is_background_task_running(project_uuid, user_id):
```

Comment:
Can comment why this is required?

Reply:
Ensures only one background task at a time deletes a given ``project_uuid``, ``user_id``) pair.

If you have multiple tasks deleting a file at most one will succeed and the rest fail. 
This avoids having a burst of background tasks that were fire&forget failing